service/src/main/java/org/ehrbase/repository/EhrFolderRepository.java

```diff
@@ -100,11 +105,37 @@ public void commit(
                 ContributionChangeType.creation,
                 r -> r.setEhrFoldersIdx(ehrFoldersIdx),
                 r -> {
+                    List<String> uuids = uuidExtract.apply(r.getData().data());
+                    if (uuids == null) r.setItemUuids(new UUID[0]);
+                    else r.setItemUuids(uuids.stream().map(UUID::fromString).toArray(UUID[]::new));
```

Comment:
Currently there is no validation for a valid items->id->value UUID. Should we first validate if the given value is a valid UUID and fail gracefully, or this this part of the service layer? Another question is what should we do in case we get a versioned object ID, should we only the id part?

Reply:
This is currently in discussionb, so I leave it open for now

Reply:
We should postpone thorough validation, but maybe throw a more descriptive IllegalArgumentException when UUID::fromString fails